lms/static/js/spec/ccx/schedule_spec.js

```diff
@@ -0,0 +1,197 @@
+define(['js/ccx/schedule'],
+    function() {
+        describe("edx.ccx.schedule.ScheduleView", function() {
+            var view = null,
+                ajaxSuccess = true;
+
+            var assertAjax = function(url, method, data) {
+                expect($.ajax).toHaveBeenCalled();
+                var ajaxArgs = $.ajax.mostRecentCall.args[0];
+                expect(ajaxArgs.url).toEqual(url);
+                expect(ajaxArgs.type).toEqual(method);
+                expect(ajaxArgs.data).toEqual(JSON.stringify(data))
+            };
+
+            beforeEach(function() {
+                loadFixtures("js/fixtures/ccx/schedule.html");
+
+                var scheduleFixture = readFixtures("templates/ccx/schedule.underscore");
+                appendSetFixtures("<div id=\"schedule_template\">" + scheduleFixture + "</div>");
+                schedule_template = _.template($('#schedule_template').html());
+                save_url = 'save_ccx';
+
+                $.fn.leanModal = function(param) {
+                    return true;
+                }
+
+                // Stub AJAX calls to return success / failure
+                spyOn($, "ajax").andCallFake(function(e) {
```

Comment:
You'd be better off using Sinon.js's [fake `XMLHttpRequest` implementation](http://sinonjs.org/docs/#server): it's cleaner and more complete. We're already using Sinon.js in other Jasmine tests, so you can copy from another test as an example, like [this one](https://github.com/edx/edx-platform/blob/master/cms/static/coffee/spec/views/textbook_spec.coffee#L86). (We're wrapping the sinon calls in a module called [ajax_helpers](https://github.com/edx/edx-platform/blob/master/common/static/js/spec_helpers/ajax_helpers.js).)